extensions/common/context.cc

```diff
@@ -193,7 +193,7 @@ std::string_view ProtocolString(Protocol protocol) {
 // Retrieves the traffic direction from the configuration context.
 TrafficDirection getTrafficDirection() {
   int64_t direction;
-  if (getValue({"listener_direction"}, &direction)) {
+  if (getValue({"xds", "listener_direction"}, &direction)) {
```

Comment:
looks like there's a bug, `xds.listener_direction` didn't return the right result, always return `TrafficDirection`.

Reply:
see https://github.com/istio/proxy/pull/5300

Reply:
Yeap, I know about this case. It doesn't actually make sense to read listener direction during config time with ECDS since the same filter instance can be used for both directions.